static/src/javascripts/projects/common/modules/gallery/lightbox.js

```diff
@@ -0,0 +1,626 @@
+            },
+            close() {
+                this.state = 'closed';
+            },
+        },
+    },
+
+    endslate: {
+        enter() {
+            this.translateContent(this.$slides.length, 0, 0);
+            this.index = this.images.length + 1;
+            mediator.on('window:throttledResize', this.resize);
+        },
+        leave() {
+            mediator.off('window:throttledResize', this.resize);
+        },
+        events: {
+            next() {
+                pulseButton(this.nextBtn);
+                this.index = 1;
+                this.state = 'image';
+            },
+            prev() {
+                pulseButton(this.prevBtn);
+                this.index = this.images.length;
+                this.state = 'image';
+            },
+            reload() {
+                this.reloadState = true;
+            },
+            resize() {
+                this.swipeContainerWidth = this.$swipeContainer.dim().width;
+                this.translateContent(this.$slides.length, 0, 0);
+            },
+            close() {
+                this.state = 'closed';
+            },
+        },
+    },
+};
+
+GalleryLightbox.prototype.endslate = new Component();
+
+const init = (): void => {
+    loadCssPromise.then(() => {
+        const images = config.get('page.lightboxImages');
+
+        if (images && images.images.length > 0) {
+            let lightbox;
+            const galleryHash = window.location.hash;
+
+            let res;
+
+            bean.on(document.body, 'click', '.js-gallerythumbs', e => {
+                e.preventDefault();
+
+                const $el = bonzo(e.currentTarget);
+                const galleryHref =
+                    $el.attr('href') || $el.attr('data-gallery-url');
+                const galleryHrefParts = galleryHref.split('#img-');
+                const parsedGalleryIndex = parseInt(galleryHrefParts[1], 10);
+                const galleryIndex = Number.isNaN(parsedGalleryIndex)
+                    ? 1
+                    : parsedGalleryIndex; // 1-based index
+                lightbox = lightbox || new GalleryLightbox();
+
+                lightbox.loadGalleryfromJson(images, galleryIndex);
+            });
+
+            lightbox = lightbox || new GalleryLightbox();
+            const galleryId = `/${config.get('page.pageId')}`;
+            const match = /\?index=(\d+)/.exec(document.location.href);
+            if (match) {
+                // index specified so launch lightbox at that index
+                pushUrl({}, document.title, galleryId, true); // lets back work properly
+                lightbox.loadGalleryfromJson(images, parseInt(match[1], 10));
+            } else {
+                res = /^#(?:img-)?(\d+)$/.exec(galleryHash);
+                if (res) {
+                    lightbox.loadGalleryfromJson(images, parseInt(res[1], 10));
+                }
+            }
+        }
+    });
+};
+
+export { init, GalleryLightbox };
```

Comment:
`GalleryLightbox` doesn't appear to be imported anywhere. Can we remove it from the `export` object?

Reply:
It is used in the tests, so I followed the advice in the wiki to export it differently